### PR TITLE
Set up linting

### PR DESCRIPTION
**Pull request by jdleesmiller**
Since this is its own repo, set up linting and formatting.

This isn't identical to our usual setup, because it uses some newer package versions, but it is relatively close. (Several defaults [flipped](https://prettier.io/docs/en/options.html#arrow-function-parentheses) in prettier 2.0.)

Also update mocha to fix a security alert.

(And I sneakily set up CI in https://github.com/overleaf/o-error/pull/8 just with Circle, since this is an open source repo.)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "extends": [
+    "plugin:prettier/recommended",
```
Comment (40thieves) [inline .eslintrc.json:3]: No `standard`?

File: README.md [hunk 2]
```diff
@@ -17,10 +18,10 @@ ES6 classes make it easy to define custom errors by subclassing `Error`. Subclas
 ```js
 const OError = require('@overleaf/o-error')
 
-function doSomethingBad () {
+function doSomethingBad() {
   throw new OError({
     message: 'did something bad',
-    info: { thing: 'foo' }
+    info: { thing: 'foo' },
```
Comment (40thieves) [inline README.md:24]: This does feel like a departure from our existing style. Is this part of the prettier update? I assumed it was standard until I noticed that we're not using that

File: test/http.test.js [hunk 3]
```diff
@@ -1,16 +1,16 @@
 const OError = require('..')
 const HttpErrors = require('../http')
 
-describe('OError/http', () => {
-  it('is instance of OError', () => {
+describe('OError/http', function () {
```
Comment (40thieves) [inline test/http.test.js:4]: After being a bit confused, looks like this has been the prettier style for a while, and it's our standard linting thats changing it?

File: README.md (hunk 2 above)
Comment (jdleesmiller) [reply README.md:24]: Yep: https://prettier.io/docs/en/options.html#trailing-commas

File: test/http.test.js (hunk 3 above)
Comment (40thieves) [reply test/http.test.js:4]: (Just realised my comment isn't clear: I meant the space before the parens)

Comment (jdleesmiller) [reply test/http.test.js:4]: That seems to be the case for anonymous functions now... even with standard.

File: .eslintrc.json (hunk 1 above)
Comment (jdleesmiller) [reply .eslintrc.json:4]: Good point, I should have added that!

**Pull request merged**